generator/src/main/java/com/scottlogic/deg/generator/Guice/GenerationEngineProvider.java

```diff
@@ -2,20 +2,21 @@
 
 import com.google.inject.Inject;
 import com.google.inject.Provider;
-import com.google.inject.name.Named;
-import com.scottlogic.deg.generator.CommandLine.GenerateCommandLine;
 import com.scottlogic.deg.generator.GenerationEngine;
+import com.scottlogic.deg.generator.StandardGenerationEngine;
+import com.scottlogic.deg.generator.ViolationGenerationEngine;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
 
 public class GenerationEngineProvider implements Provider<GenerationEngine> {
 
-    private GenerateCommandLine commandLine;
+    private GenerationConfigSource commandLine;
     private GenerationEngine generationEngine;
     private GenerationEngine violationEngine;
 
     @Inject
-    public GenerationEngineProvider(GenerateCommandLine commandLine,
-                                    @Named("valid") GenerationEngine generationEngine,
-                                    @Named("invalid") GenerationEngine violationEngine) {
+    public GenerationEngineProvider(GenerationConfigSource commandLine,
+                                    StandardGenerationEngine generationEngine,
+                                    ViolationGenerationEngine violationEngine) {
```

Comment:
Discussion point: Do we want to change this to implicit binding, or use `@Named`?

Reply:
see #495 for extra info

Reply:
The `@named` wasn't really giving any benefits, in fact it looks quite weird to say your class can accept two generic GenerationEngines - but only if they are "named" correctly. Since what you actually mean is the provider needs two different generation engines this is cleaner.